cmd/admissioncontroller/handlers.go

```diff
@@ -0,0 +1,379 @@
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "apps"
+	podschema.Version = "v1"
+	podschema.Kind = "DaemonSet"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*extensions.DaemonSet)
+
+	pod := new(core.Pod)
+	ds.Spec.Template.Spec.DeepCopyInto(&pod.Spec)
+	ds.Spec.Template.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	ac.checkPodSpec(pod, ds.Namespace, w, req.UID)
+}
+
+func (ac *admissionController) handleReplicaSet(w http.ResponseWriter, r *http.Request) {
+	req, errcode := getAdmissionReviewRequest(r)
+	if errcode != 0 {
+		http.Error(w, http.StatusText(http.StatusUnsupportedMediaType), http.StatusUnsupportedMediaType)
+		return
+	}
+	if req.UID == "" ||
+		req.Kind.Group != "apps" || req.Kind.Version != "v1" || req.Kind.Kind != "ReplicaSet" ||
+		req.Resource.Group != "apps" || req.Resource.Version != "v1" || req.Resource.Resource != "replicasets" ||
+		req.SubResource != "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		log.Printf("Field mismatch")
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "apps"
+	podschema.Version = "v1"
+	podschema.Kind = "ReplicaSet"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*extensions.ReplicaSet)
+
+	pod := new(core.Pod)
+	ds.Spec.Template.Spec.DeepCopyInto(&pod.Spec)
+	ds.Spec.Template.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	ac.checkPodSpec(pod, ds.Namespace, w, req.UID)
+}
+
+func (ac *admissionController) handleStatefulSet(w http.ResponseWriter, r *http.Request) {
+	req, errcode := getAdmissionReviewRequest(r)
+	if errcode != 0 {
+		http.Error(w, http.StatusText(http.StatusUnsupportedMediaType), http.StatusUnsupportedMediaType)
+		return
+	}
+	if req.UID == "" ||
+		req.Kind.Group != "apps" || req.Kind.Version != "v1" || req.Kind.Kind != "StatefulSet" ||
+		req.Resource.Group != "apps" || req.Resource.Version != "v1" || req.Resource.Resource != "statefulsets" ||
+		req.SubResource != "" {
+		http.Error(w, http.StatusText(http.StatusBadRequest), http.StatusBadRequest)
+		log.Printf("Field mismatch")
+		return
+	}
+
+	var podschema schema.GroupVersionKind
+	podschema.Group = "apps"
+	podschema.Version = "v1"
+	podschema.Kind = "StatefulSet"
+	o, _, err := codec.Decode(req.Object.Raw, &podschema, nil)
+	if err != nil {
+		log.Printf("Decode error:  %s", err)
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+	ds := o.(*apps.StatefulSet)
+
+	pod := new(core.Pod)
+	ds.Spec.Template.Spec.DeepCopyInto(&pod.Spec)
+	ds.Spec.Template.ObjectMeta.DeepCopyInto(&pod.ObjectMeta)
+	ac.checkPodSpec(pod, ds.Namespace, w, req.UID)
+}
+
+func (ac *admissionController) handleJob(w http.ResponseWriter, r *http.Request) {
```

Comment:
1) there is a lot of repetition in these functions - try to boil it down
2) some objects are available through multiple groups - e.g. daemonsets via apps and extensions - need to check this